packages/fmg-core/contracts/Rules.sol

```diff
@@ -28,156 +28,277 @@ library Rules {
+                    "Invalid transition from PostFundSetup: resolutions must be equal"
+                );
             } else {
                 // PostFundSetup -> Conclude
-                require(_toState.stateType() == State.StateType.Conclude);
-                require(State.resolutionsEqual(_fromState, _toState));
+                require(
+                    _toState.isConclude(),
+                    "Invalid transition from PostFundSetup: stateType must be Conclude"
+                );
+                require(
+                    State.resolutionsEqual(_fromState, _toState),
+                    "Invalid transition from PostFundSetup: resolutions must be equal"
+                );
             }
         }
         return true;
     }
 
-    function validTransitionFromGame(bytes _fromState, bytes _toState) public pure returns (bool) {
-        if (_toState.stateType() == State.StateType.Game) {
-            require(ForceMoveGame(_fromState.channelType()).validTransition(_fromState, _toState));
+    function validTransitionFromGame(
+        State.StateStruct memory _fromState,
+        State.StateStruct memory _toState
+    ) public pure returns (bool) {
+        if (_toState.isGame()) {
+            require(
+                validGameTransition(_fromState, _toState),
+                "Invalid transition from Game: transition must be valid"
+            );
         } else {
-            require(_toState.stateType() == State.StateType.Conclude);
-            require(State.resolutionsEqual(_fromState, _toState));
+            require(
+                _toState.isConclude(),
+                "Invalid transition from Game: stateType must be Conclude"
+            );
+            require(
+                State.resolutionsEqual(_fromState, _toState),
+                "Invalid transition from Game: resolutions must be equal"
+            );
         }
         return true;
     }
 
-    function validTransitionFromConclude(bytes _fromState, bytes _toState) public pure returns (bool) {
-        require(_toState.stateType() == State.StateType.Conclude);
-        require(State.resolutionsEqual(_fromState, _toState));
+    function validTransitionFromConclude(
+        State.StateStruct memory _fromState,
+        State.StateStruct memory _toState
+    ) public pure returns (bool) {
+        require(
+            _toState.isConclude(),
+            "Invalid transition from Conclude: stateType must be Conclude"
+        );
+        require(
+            State.resolutionsEqual(_fromState, _toState),
+            "Invalid transition from Conclude: resolutions must be equal"
+        );
+        return true;
+    }
+
+    function validGameTransition(
+        State.StateStruct memory _fromState,
+        State.StateStruct memory _toState
+    ) public pure returns (bool) {
         return true;
+        // return ForceMoveGame(_fromState.channelType).validTransition(_fromState.gameAttributes, _toState.gameAttributes);
```

Comment:
`validGameTransition` was included here for the sake of [L266](https://github.com/magmo/force-move-games/pull/43/files#diff-3b4aa17b465a96afb1ddc5bf4bbca4f3R266).

I'll add a test of this using the `CountingGame`.

I can do that in this PR, or as a separate unit of work.